upup/pkg/fi/cloudup/hetznertasks/servergroup.go

```diff
@@ -143,6 +148,28 @@ func (_ *ServerGroup) CheckChanges(a, e, changes *ServerGroup) error {
 func (_ *ServerGroup) RenderHetzner(t *hetzner.HetznerAPITarget, a, e, changes *ServerGroup) error {
 	client := t.Cloud.ServerClient()
 
+	if a != nil {
+		// Add "kops.k8s.io/needs-update" label to servers needing update
+		for _, serverName := range a.NeedUpdate {
+			server, _, err := client.GetByName(context.TODO(), serverName)
+			if err != nil {
+				return err
+			}
+			if server == nil {
+				continue
+			}
+
+			server.Labels[hetzner.TagKubernetesInstanceNeedsUpdate] = ""
+			_, _, err = client.Update(context.TODO(), server, hcloud.ServerUpdateOpts{
+				Name:   server.Name,
+				Labels: server.Labels,
+			})
+			if err != nil {
+				return err
```

Comment:
Nit: when we get an error from calling the hetzner API (or any API), I don't know how good those errors are.  It might be nice to `fmt.Errorf("error setting needs-update label on %q: %w", serverName, err)` for example.